ccl/backupccl: TestProtectedTimestampSpanSelectionDuringBackup failed

[(ccl/backupccl).TestProtectedTimestampSpanSelectionDuringBackup failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2495086&tab=buildLog) on [master@aa69b42032f89d1bb777986ff7a8d4f753b71d83](https://github.com/cockroachdb/cockroach/commits/aa69b42032f89d1bb777986ff7a8d4f753b71d83):

```
=== RUN   TestProtectedTimestampSpanSelectionDuringBackup
    test_log_scope.go:72: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestProtectedTimestampSpanSelectionDuringBackup898532731
    test_log_scope.go:73: use -show-logs to present logs inline
n1: found liveness
n2: found liveness
n3: found liveness
=== CONT  TestProtectedTimestampSpanSelectionDuringBackup
    backup_test.go:5864: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestProtectedTimestampSpanSelectionDuringBackup898532731
--- FAIL: TestProtectedTimestampSpanSelectionDuringBackup (17.40s)
=== RUN   TestProtectedTimestampSpanSelectionDuringBackup/last-index-gced
E201204 06:21:25.309528 521981 kv/kvserver/queue.go:1087  [n1,split,s1,r46/1:‹/{Table/69-Max}›] unable to split [n1,s1,r46/1:/{Table/69-Max}] at key "/Table/70": split at key /Table/70 failed: batch timestamp 1607062884.203095465,0 must be after replica GC threshold 1607062884.307758898,0
    backup_test.go:5855: error executing 'DROP INDEX foo@baz': pq: batch timestamp 1607062884.286029702,0 must be after replica GC threshold 1607062884.308103221,0
E201204 06:21:29.212540 519781 jobs/adopt.go:255  [n1] job 612736962438332417: adoption completed with error job 612736962438332417: node liveness error: restarting in background
E201204 06:21:29.212671 519537 jobs/adopt.go:255  [n1] job 612736962304737281: adoption completed with error job 612736962304737281: node liveness error: restarting in background
E201204 06:21:29.212749 521388 jobs/adopt.go:255  [n1] job 612736969564782593: adoption completed with error job 612736969564782593: node liveness error: restarting in background
E201204 06:21:29.212762 521261 jobs/adopt.go:255  [n1] job 612736968572829697: adoption completed with error job 612736968572829697: node liveness error: restarting in background
E201204 06:21:29.212793 520996 jobs/adopt.go:255  [n1] job 612736966384353281: adoption completed with error job 612736966384353281: node liveness error: restarting in background
E201204 06:21:29.212765 521402 jobs/adopt.go:255  [n1] job 612736971736416257: adoption completed with error job 612736971736416257: node liveness error: restarting in background
E201204 06:21:29.212657 520253 jobs/adopt.go:255  [n1] job 612736963711795201: adoption completed with error job 612736963711795201: node liveness error: restarting in background
E201204 06:21:29.212834 513991 jobs/adopt.go:255  [n1] job 612736947589316609: adoption completed with error job 612736947589316609: node liveness error: restarting in background
E201204 06:21:29.212862 514376 jobs/adopt.go:255  [n1] job 612736949860794369: adoption completed with error job 612736949860794369: node liveness error: restarting in background
E201204 06:21:29.212760 520544 jobs/adopt.go:255  [n1] job 612736965428379649: adoption completed with error job 612736965428379649: node liveness error: restarting in background
E201204 06:21:29.212895 511854 jobs/adopt.go:255  [n1] job 612736946318671873: adoption completed with error job 612736946318671873: node liveness error: restarting in background
    --- FAIL: TestProtectedTimestampSpanSelectionDuringBackup/last-index-gced (5.08s)

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestProtectedTimestampSpanSelectionDuringBackup PKG=./pkg/ccl/backupccl TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestProtectedTimestampSpanSelectionDuringBackup.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(ccl/backupccl).TestProtectedTimestampSpanSelectionDuringBackup failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2504739&tab=buildLog) on [master@2807d09091584a0224c2a6f9c9ec99db6fe490c6](https://github.com/cockroachdb/cockroach/commits/2807d09091584a0224c2a6f9c9ec99db6fe490c6):

```
=== RUN   TestProtectedTimestampSpanSelectionDuringBackup
    test_log_scope.go:72: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestProtectedTimestampSpanSelectionDuringBackup629060205
    test_log_scope.go:73: use -show-logs to present logs inline
n1: found liveness
n2: found liveness
n3: found liveness
=== CONT  TestProtectedTimestampSpanSelectionDuringBackup
    backup_test.go:5880: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestProtectedTimestampSpanSelectionDuringBackup629060205
--- FAIL: TestProtectedTimestampSpanSelectionDuringBackup (44.16s)
=== RUN   TestProtectedTimestampSpanSelectionDuringBackup/last-index-gced
    backup_test.go:5876: 
        	Error Trace:	backup_test.go:5876
        	Error:      	Not equal: 
        	            	expected: []string{"/Table/69/{1-2}", "/Table/70/{1-3}"}
        	            	actual  : []string{"/Table/70/{1-2}", "/Table/71/{1-3}"}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1,4 +1,4 @@
        	            	 ([]string) (len=2) {
        	            	- (string) (len=15) "/Table/69/{1-2}",
        	            	- (string) (len=15) "/Table/70/{1-3}"
        	            	+ (string) (len=15) "/Table/70/{1-2}",
        	            	+ (string) (len=15) "/Table/71/{1-3}"
        	            	 }
        	Test:       	TestProtectedTimestampSpanSelectionDuringBackup/last-index-gced
E201208 09:40:10.865236 383568 kv/kvclient/kvcoord/txn_interceptor_committer.go:449  making txn commit explicit failed for "unnamed" meta={id=388dc996 pri=0.00780537 epo=0 ts=1607420410.689466501,0 min=1607420410.689466501,0 seq=6} lock=true stat=STAGING rts=1607420410.689466501,0 wto=false max=1607420410.689466501,0 ifw=5: result is ambiguous (server shutdown)
    --- FAIL: TestProtectedTimestampSpanSelectionDuringBackup/last-index-gced (7.68s)

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestProtectedTimestampSpanSelectionDuringBackup PKG=./pkg/ccl/backupccl TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestProtectedTimestampSpanSelectionDuringBackup.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(ccl/backupccl).TestProtectedTimestampSpanSelectionDuringBackup failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2505268&tab=buildLog) on [master@856172f0b0e848b5468d7cbe6954e46a9af6ba32](https://github.com/cockroachdb/cockroach/commits/856172f0b0e848b5468d7cbe6954e46a9af6ba32):

```
=== RUN   TestProtectedTimestampSpanSelectionDuringBackup
    test_log_scope.go:72: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestProtectedTimestampSpanSelectionDuringBackup824856458
    test_log_scope.go:73: use -show-logs to present logs inline
n1: found liveness
n2: found liveness
n3: found liveness
=== CONT  TestProtectedTimestampSpanSelectionDuringBackup
    backup_test.go:5879: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestProtectedTimestampSpanSelectionDuringBackup824856458
--- FAIL: TestProtectedTimestampSpanSelectionDuringBackup (43.07s)
=== RUN   TestProtectedTimestampSpanSelectionDuringBackup/last-index-dropped
    backup_test.go:5859: 
        	Error Trace:	backup_test.go:5859
        	Error:      	Not equal: 
        	            	expected: []string{"/Table/66/{1-2}", "/Table/67/{1-3}"}
        	            	actual  : []string{"/Table/67/{1-2}", "/Table/68/{1-3}"}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1,4 +1,4 @@
        	            	 ([]string) (len=2) {
        	            	- (string) (len=15) "/Table/66/{1-2}",
        	            	- (string) (len=15) "/Table/67/{1-3}"
        	            	+ (string) (len=15) "/Table/67/{1-2}",
        	            	+ (string) (len=15) "/Table/68/{1-3}"
        	            	 }
        	Test:       	TestProtectedTimestampSpanSelectionDuringBackup/last-index-dropped
    --- FAIL: TestProtectedTimestampSpanSelectionDuringBackup/last-index-dropped (5.44s)
=== RUN   TestProtectedTimestampSpanSelectionDuringBackup/last-index-gced
    backup_test.go:5865: error executing 'CREATE DATABASE test; USE test;': pq: database "test" already exists
E201208 18:47:02.268439 367256 jobs/registry.go:642  failed to serve pause and cancel requests: could not query jobs table: cancel/pause-requested: context canceled
E201208 18:47:02.274436 385867 kv/kvclient/kvcoord/txn_interceptor_committer.go:449  making txn commit explicit failed for "unnamed" meta={id=1a9afce8 pri=0.08733743 epo=0 ts=1607453222.141090219,1 min=1607453222.028745896,0 seq=6} lock=true stat=STAGING rts=1607453222.141090219,1 wto=false max=1607453222.118729627,0 ifw=5: result is ambiguous (error=failed to connect to n2 at 127.0.0.1:38353: context canceled [exhausted])
    --- FAIL: TestProtectedTimestampSpanSelectionDuringBackup/last-index-gced (0.00s)

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestProtectedTimestampSpanSelectionDuringBackup PKG=./pkg/ccl/backupccl TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestProtectedTimestampSpanSelectionDuringBackup.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(ccl/backupccl).TestProtectedTimestampSpanSelectionDuringBackup failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2521125&tab=buildLog) on [master@d06d00f5e8f85dd2cd581a36afb35f9c57a774a1](https://github.com/cockroachdb/cockroach/commits/d06d00f5e8f85dd2cd581a36afb35f9c57a774a1):

```
=== RUN   TestProtectedTimestampSpanSelectionDuringBackup
    test_log_scope.go:72: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestProtectedTimestampSpanSelectionDuringBackup270781309
    test_log_scope.go:73: use -show-logs to present logs inline
n1: found liveness
n2: found liveness
n3: found liveness
=== CONT  TestProtectedTimestampSpanSelectionDuringBackup
    backup_test.go:5881: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestProtectedTimestampSpanSelectionDuringBackup270781309
--- FAIL: TestProtectedTimestampSpanSelectionDuringBackup (39.62s)
=== RUN   TestProtectedTimestampSpanSelectionDuringBackup/last-index-dropped
    backup_test.go:5860: 
        	Error Trace:	backup_test.go:5860
        	Error:      	Not equal: 
        	            	expected: []string{"/Table/66/{1-2}", "/Table/67/{1-3}"}
        	            	actual  : []string{"/Table/66/{1-2}", "/Table/68/{1-3}"}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -2,3 +2,3 @@
        	            	  (string) (len=15) "/Table/66/{1-2}",
        	            	- (string) (len=15) "/Table/67/{1-3}"
        	            	+ (string) (len=15) "/Table/68/{1-3}"
        	            	 }
        	Test:       	TestProtectedTimestampSpanSelectionDuringBackup/last-index-dropped
    --- FAIL: TestProtectedTimestampSpanSelectionDuringBackup/last-index-dropped (5.38s)

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestProtectedTimestampSpanSelectionDuringBackup PKG=./pkg/ccl/backupccl TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestProtectedTimestampSpanSelectionDuringBackup.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(ccl/backupccl).TestProtectedTimestampSpanSelectionDuringBackup failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2537500&tab=buildLog) on [master@d314ad9ea31b72fcf9797088d790ff8087a42603](https://github.com/cockroachdb/cockroach/commits/d314ad9ea31b72fcf9797088d790ff8087a42603):

```
=== RUN   TestProtectedTimestampSpanSelectionDuringBackup
    test_log_scope.go:72: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestProtectedTimestampSpanSelectionDuringBackup150312148
    test_log_scope.go:73: use -show-logs to present logs inline
n1: found liveness
n2: found liveness
n3: found liveness
=== CONT  TestProtectedTimestampSpanSelectionDuringBackup
    backup_test.go:5912: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestProtectedTimestampSpanSelectionDuringBackup150312148
--- FAIL: TestProtectedTimestampSpanSelectionDuringBackup (37.54s)
=== RUN   TestProtectedTimestampSpanSelectionDuringBackup/last-index-dropped
    backup_test.go:5891: 
        	Error Trace:	backup_test.go:5891
        	Error:      	Not equal: 
        	            	expected: []string{"/Table/66/{1-2}", "/Table/67/{1-3}"}
        	            	actual  : []string{"/Table/67/{1-2}", "/Table/68/{1-3}"}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1,4 +1,4 @@
        	            	 ([]string) (len=2) {
        	            	- (string) (len=15) "/Table/66/{1-2}",
        	            	- (string) (len=15) "/Table/67/{1-3}"
        	            	+ (string) (len=15) "/Table/67/{1-2}",
        	            	+ (string) (len=15) "/Table/68/{1-3}"
        	            	 }
        	Test:       	TestProtectedTimestampSpanSelectionDuringBackup/last-index-dropped
    --- FAIL: TestProtectedTimestampSpanSelectionDuringBackup/last-index-dropped (3.87s)

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestProtectedTimestampSpanSelectionDuringBackup PKG=./pkg/ccl/backupccl TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestProtectedTimestampSpanSelectionDuringBackup.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(ccl/backupccl).TestProtectedTimestampSpanSelectionDuringBackup failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2556566&tab=buildLog) on [master@fb71ee4a4243e632b3f8589d45ca65efa6f68991](https://github.com/cockroachdb/cockroach/commits/fb71ee4a4243e632b3f8589d45ca65efa6f68991):

```
=== RUN   TestProtectedTimestampSpanSelectionDuringBackup
    test_log_scope.go:72: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestProtectedTimestampSpanSelectionDuringBackup850521782
    test_log_scope.go:73: use -show-logs to present logs inline
n1: found liveness
n2: found liveness
n3: found liveness
=== CONT  TestProtectedTimestampSpanSelectionDuringBackup
    backup_test.go:5912: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestProtectedTimestampSpanSelectionDuringBackup850521782
--- FAIL: TestProtectedTimestampSpanSelectionDuringBackup (35.42s)
=== RUN   TestProtectedTimestampSpanSelectionDuringBackup/last-index-dropped
    backup_test.go:5891: 
        	Error Trace:	backup_test.go:5891
        	Error:      	Not equal: 
        	            	expected: []string{"/Table/66/{1-2}", "/Table/67/{1-3}"}
        	            	actual  : []string{"/Table/67/{1-2}", "/Table/68/{1-3}"}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1,4 +1,4 @@
        	            	 ([]string) (len=2) {
        	            	- (string) (len=15) "/Table/66/{1-2}",
        	            	- (string) (len=15) "/Table/67/{1-3}"
        	            	+ (string) (len=15) "/Table/67/{1-2}",
        	            	+ (string) (len=15) "/Table/68/{1-3}"
        	            	 }
        	Test:       	TestProtectedTimestampSpanSelectionDuringBackup/last-index-dropped
    --- FAIL: TestProtectedTimestampSpanSelectionDuringBackup/last-index-dropped (3.95s)

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestProtectedTimestampSpanSelectionDuringBackup PKG=./pkg/ccl/backupccl TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestProtectedTimestampSpanSelectionDuringBackup.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(ccl/backupccl).TestProtectedTimestampSpanSelectionDuringBackup failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2736861&tab=buildLog) on [master@bec3bd044a3b87be367a92c1eafa9e4183a624e4](https://github.com/cockroachdb/cockroach/commits/bec3bd044a3b87be367a92c1eafa9e4183a624e4):

```
=== RUN   TestProtectedTimestampSpanSelectionDuringBackup
    test_log_scope.go:73: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestProtectedTimestampSpanSelectionDuringBackup089203619
    test_log_scope.go:74: use -show-logs to present logs inline
n1: found liveness
n2: found liveness
n3: found liveness
=== CONT  TestProtectedTimestampSpanSelectionDuringBackup
    backup_test.go:5950: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestProtectedTimestampSpanSelectionDuringBackup089203619
--- FAIL: TestProtectedTimestampSpanSelectionDuringBackup (49.08s)
=== RUN   TestProtectedTimestampSpanSelectionDuringBackup/last-index-dropped
    backup_test.go:5922: error executing 'CREATE DATABASE test; USE test;': pq: database "test" already exists
    --- FAIL: TestProtectedTimestampSpanSelectionDuringBackup/last-index-dropped (0.01s)
=== RUN   TestProtectedTimestampSpanSelectionDuringBackup/interleaved-spans
    backup_test.go:5883: 
        	Error Trace:	backup_test.go:5883
        	Error:      	Not equal: 
        	            	expected: []string{"/Table/59/{1-3}", "/Table/61/{2-3}"}
        	            	actual  : []string{"/Table/59/{1-3}", "/Table/62/{2-3}"}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -2,3 +2,3 @@
        	            	  (string) (len=15) "/Table/59/{1-3}",
        	            	- (string) (len=15) "/Table/61/{2-3}"
        	            	+ (string) (len=15) "/Table/62/{2-3}"
        	            	 }
        	Test:       	TestProtectedTimestampSpanSelectionDuringBackup/interleaved-spans
    --- FAIL: TestProtectedTimestampSpanSelectionDuringBackup/interleaved-spans (5.84s)
=== RUN   TestProtectedTimestampSpanSelectionDuringBackup/revs-span-merge
    backup_test.go:5889: error executing 'CREATE DATABASE test; USE test;': pq: database "test" already exists
    --- FAIL: TestProtectedTimestampSpanSelectionDuringBackup/revs-span-merge (0.02s)

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestProtectedTimestampSpanSelectionDuringBackup PKG=./pkg/ccl/backupccl TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestProtectedTimestampSpanSelectionDuringBackup.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

--- a/pkg/ccl/backupccl/backup_test.go
+++ b/pkg/ccl/backupccl/backup_test.go
@@ -5790,6 +5790,11 @@ func TestProtectedTimestampsDuringBackup(t *testing.T) {
 	}
 }
 
+func getTableID(db *kv.DB, dbName, tableName string) descpb.ID {
+	desc := catalogkv.TestingGetTableDescriptor(db, keys.SystemSQLCodec, dbName, tableName)
+	return desc.GetID()
+}
+
 // TestSpanSelectionDuringBackup tests the method spansForAllTableIndexes which
 // is used to resolve the spans which will be backed up, and spans for which
 // protected ts records will be created.
@@ -5824,6 +5829,7 @@ func TestProtectedTimestampSpanSelectionDuringBackup(t *testing.T) {
 
 	conn := tc.ServerConn(0)
 	runner := sqlutils.MakeSQLRunner(conn)
+	db := tc.Server(0).DB()
 	baseBackupURI := "nodelocal://0/foo/"
 
 	t.Run("contiguous-span-merge", func(t *testing.T) {
@@ -5832,7 +5838,8 @@ func TestProtectedTimestampSpanSelectionDuringBackup(t *testing.T) {
 			"INDEX baz(name), INDEX bar (v))")
 
 		runner.Exec(t, fmt.Sprintf(`BACKUP DATABASE test INTO '%s'`, baseBackupURI+t.Name()))
-		require.Equal(t, []string{"/Table/53/{1-4}"}, actualResolvedSpans)
+		tableID := getTableID(db, "test", "foo")
+		require.Equal(t, []string{fmt.Sprintf("/Table/%d/{1-4}", tableID)}, actualResolvedSpans)
 		runner.Exec(t, "DROP DATABASE test;")
 		actualResolvedSpans = nil
 	})
@@ -5846,7 +5853,9 @@ func TestProtectedTimestampSpanSelectionDuringBackup(t *testing.T) {
 		runner.Exec(t, "DROP INDEX foo@baz")
 
 		runner.Exec(t, fmt.Sprintf(`BACKUP DATABASE test INTO '%s'`, baseBackupURI+t.Name()))
-		require.Equal(t, []string{"/Table/55/{1-2}", "/Table/55/{3-4}"}, actualResolvedSpans)
+		tableID := getTableID(db, "test", "foo")
+		require.Equal(t, []string{fmt.Sprintf("/Table/%d/{1-2}", tableID),
+			fmt.Sprintf("/Table/%d/{3-4}", tableID)}, actualResolvedSpans)
 		runner.Exec(t, "DROP DATABASE test;")
 		actualResolvedSpans = nil
 	})
@@ -5861,13 +5870,13 @@ func TestProtectedTimestampSpanSelectionDuringBackup(t *testing.T) {
 		time.Sleep(time.Second * 2)
 
 		runner.Exec(t, fmt.Sprintf(`BACKUP DATABASE test INTO '%s'`, baseBackupURI+t.Name()))
-		require.Equal(t, []string{"/Table/57/{1-4}"}, actualResolvedSpans)
+		tableID := getTableID(db, "test", "foo")
+		require.Equal(t, []string{fmt.Sprintf("/Table/%d/{1-4}", tableID)}, actualResolvedSpans)
 		runner.Exec(t, "DROP DATABASE test;")
 		actualResolvedSpans = nil
 	})
 
 	t.Run("interleaved-spans", func(t *testing.T) {
-		skip.WithIssue(t, 57546, "flaky test")
 		runner.Exec(t, "CREATE DATABASE test; USE test;")
 		runner.Exec(t, "CREATE TABLE grandparent (a INT PRIMARY KEY, v BYTES, INDEX gpindex (v))")
 		runner.Exec(t, "CREATE TABLE parent (a INT, b INT, v BYTES, "+
@@ -5880,13 +5889,15 @@ func TestProtectedTimestampSpanSelectionDuringBackup(t *testing.T) {
 		// tables parent and child.
 		// /Table/59/2 - /Table/59/3 is for the gpindex
 		// /Table/61/{2-3} is for the childindex
-		require.Equal(t, []string{"/Table/59/{1-3}", "/Table/61/{2-3}"}, actualResolvedSpans)
+		grandparentID := getTableID(db, "test", "grandparent")
+		childID := getTableID(db, "test", "child")
+		require.Equal(t, []string{fmt.Sprintf("/Table/%d/{1-3}", grandparentID),
+			fmt.Sprintf("/Table/%d/{2-3}", childID)}, actualResolvedSpans)
 		runner.Exec(t, "DROP DATABASE test;")
 		actualResolvedSpans = nil
 	})
 
 	t.Run("revs-span-merge", func(t *testing.T) {
-		skip.WithIssue(t, 57546, "flaky test")
 		runner.Exec(t, "CREATE DATABASE test; USE test;")
 		runner.Exec(t, "CREATE TABLE foo (k INT PRIMARY KEY, v BYTES, name STRING, "+
 			"INDEX baz(name), INDEX bar (v))")
@@ -5899,7 +5910,8 @@ func TestProtectedTimestampSpanSelectionDuringBackup(t *testing.T) {
 		// The BACKUP with revision history will pickup the dropped index baz as
 		// well because it existed in a non-drop state at some point in the interval
 		// covered by this BACKUP.
-		require.Equal(t, []string{"/Table/63/{1-4}"}, actualResolvedSpans)
+		tableID := getTableID(db, "test", "foo")
+		require.Equal(t, []string{fmt.Sprintf("/Table/%d/{1-4}", tableID)}, actualResolvedSpans)
 		actualResolvedSpans = nil
 		runner.Exec(t, "DROP TABLE foo")
 
@@ -5914,13 +5926,15 @@ func TestProtectedTimestampSpanSelectionDuringBackup(t *testing.T) {
 		// incremental backup with revision history. We also expect to see both drop
 		// and non-drop indexes of table foo2 as all the indexes were live at some
 		// point in the interval covered by this BACKUP.
-		require.Equal(t, []string{"/Table/63/{1-2}", "/Table/63/{3-4}", "/Table/64/{1-4}"}, actualResolvedSpans)
+		tableID2 := getTableID(db, "test", "foo2")
+		require.Equal(t, []string{fmt.Sprintf("/Table/%d/{1-2}", tableID),
+			fmt.Sprintf("/Table/%d/{3-4}", tableID), fmt.Sprintf("/Table/%d/{1-4}", tableID2)},
+			actualResolvedSpans)
 		runner.Exec(t, "DROP DATABASE test;")
 		actualResolvedSpans = nil
 	})
 
 	t.Run("last-index-dropped", func(t *testing.T) {
-		skip.WithIssue(t, 57546, "flaky test")
 		runner.Exec(t, "CREATE DATABASE test; USE test;")
 		runner.Exec(t, "CREATE TABLE foo (k INT PRIMARY KEY, v BYTES, name STRING, INDEX baz(name))")
 		runner.Exec(t, "CREATE TABLE foo2 (k INT PRIMARY KEY, v BYTES, name STRING, INDEX baz(name))")
@@ -5928,13 +5942,15 @@ func TestProtectedTimestampSpanSelectionDuringBackup(t *testing.T) {
 		runner.Exec(t, "DROP INDEX foo@baz")
 
 		runner.Exec(t, fmt.Sprintf(`BACKUP DATABASE test INTO '%s'`, baseBackupURI+t.Name()))
-		require.Equal(t, []string{"/Table/66/{1-2}", "/Table/67/{1-3}"}, actualResolvedSpans)
+		tableID := getTableID(db, "test", "foo")
+		tableID2 := getTableID(db, "test", "foo2")
+		require.Equal(t, []string{fmt.Sprintf("/Table/%d/{1-2}", tableID),
+			fmt.Sprintf("/Table/%d/{1-3}", tableID2)}, actualResolvedSpans)
 		runner.Exec(t, "DROP DATABASE test;")
 		actualResolvedSpans = nil
 	})
 
 	t.Run("last-index-gced", func(t *testing.T) {
-		skip.WithIssue(t, 57546, "flaky test")
 		runner.Exec(t, "CREATE DATABASE test; USE test;")
 		runner.Exec(t, "CREATE TABLE foo (k INT PRIMARY KEY, v BYTES, name STRING, INDEX baz(name))")
 		runner.Exec(t, "INSERT INTO foo VALUES (1, NULL, 'test')")
@@ -5945,7 +5961,10 @@ func TestProtectedTimestampSpanSelectionDuringBackup(t *testing.T) {
 		runner.Exec(t, "ALTER TABLE foo CONFIGURE ZONE USING gc.ttlseconds=60")
 
 		runner.Exec(t, fmt.Sprintf(`BACKUP DATABASE test INTO '%s'`, baseBackupURI+t.Name()))
-		require.Equal(t, []string{"/Table/69/{1-2}", "/Table/70/{1-3}"}, actualResolvedSpans)
+		tableID := getTableID(db, "test", "foo")
+		tableID2 := getTableID(db, "test", "foo2")
+		require.Equal(t, []string{fmt.Sprintf("/Table/%d/{1-2}", tableID),
+			fmt.Sprintf("/Table/%d/{1-3}", tableID2)}, actualResolvedSpans)
 		runner.Exec(t, "DROP DATABASE test;")
 		actualResolvedSpans = nil
 	})

